src/main/io/vtx_control.c

```diff
@@ -115,49 +115,53 @@ void vtxUpdateActivatedChannel(void)
 
 void vtxCycleBandOrChannel(const uint8_t bandStep, const uint8_t channelStep)
 {
-    uint8_t band = 0, channel = 0;
-    vtxDeviceCapability_t capability;
+    if (vtxCommonDeviceRegistered()) {
+        uint8_t band = 0, channel = 0;
+        vtxDeviceCapability_t capability;
 
-    bool haveAllNeededInfo = vtxCommonGetBandAndChannel(&band, &channel) && vtxCommonGetDeviceCapability(&capability);
-    if (!haveAllNeededInfo) {
-        return;
-    }
+        bool haveAllNeededInfo = vtxCommonGetBandAndChannel(&band, &channel) && vtxCommonGetDeviceCapability(&capability);
+        if (!haveAllNeededInfo) {
+            return;
+        }
 
-    int newChannel = channel + channelStep;
-    if (newChannel > capability.channelCount) {
-        newChannel = 1;
-    } else if (newChannel < 1) {
-        newChannel = capability.channelCount;
-    }
+        int newChannel = channel + channelStep;
+        if (newChannel > capability.channelCount) {
+            newChannel = 1;
+        } else if (newChannel < 1) {
+            newChannel = capability.channelCount;
+        }
 
-    int newBand = band + bandStep;
-    if (newBand > capability.bandCount) {
-        newBand = 1;
-    } else if (newBand < 1) {
-        newBand = capability.bandCount;
-    }
+        int newBand = band + bandStep;
+        if (newBand > capability.bandCount) {
+            newBand = 1;
+        } else if (newBand < 1) {
+            newBand = capability.bandCount;
+        }
 
-    vtxCommonSetBandAndChannel(newBand, newChannel);
+        vtxCommonSetBandAndChannel(newBand, newChannel);
+    }
 }
 
 void vtxCyclePower(const uint8_t powerStep)
 {
-    uint8_t power = 0;
-    vtxDeviceCapability_t capability;
+    if (vtxCommonDeviceRegistered()) {
+        uint8_t power = 0;
+        vtxDeviceCapability_t capability;
 
-    bool haveAllNeededInfo = vtxCommonGetPowerIndex(&power) && vtxCommonGetDeviceCapability(&capability);
-    if (!haveAllNeededInfo) {
-        return;
-    }
+        bool haveAllNeededInfo = vtxCommonGetPowerIndex(&power) && vtxCommonGetDeviceCapability(&capability);
+        if (!haveAllNeededInfo) {
+            return;
+        }
 
-    int newPower = power + powerStep;
-    if (newPower >= capability.powerCount) {
-        newPower = 0;
-    } else if (newPower < 0) {
-        newPower = capability.powerCount;
-    }
+        int newPower = power + powerStep;
+        if (newPower >= capability.powerCount) {
+            newPower = 0;
+        } else if (newPower < 0) {
+            newPower = capability.powerCount;
+        }
 
-    vtxCommonSetPowerByIndex(newPower);
+        vtxCommonSetPowerByIndex(newPower);
```

Comment:
Why don't you just change the configuration here and let vtxCommonProcess() do the actual work of applying the settings?